packages/databyss-editor/state/reducer.ts

```diff
@@ -74,13 +76,46 @@ export default (
     draft => {
       draft.operations = []
       draft.preventDefault = false
+      draft.resetState = false
 
       const { payload } = action
 
       // default nextSelection to `payload.selection` (which may be undef)
-      const nextSelection = payload.selection
+      let nextSelection = payload.selection
 
       switch (action.type) {
+        case PASTE: {
+          const _frag = payload.data
+
+          if (isSelectionCollapsed(state.selection)) {
+            const _isCurrentBlockEmpty = !state.blocks[
+              state.selection.anchor.index
+            ].text.textValue.length
+
+            const _spliceIndex = _isCurrentBlockEmpty
+              ? state.selection.anchor.index
+              : state.selection.anchor.index + 1
+
+            // insert block at index
+            draft.blocks.splice(
+              _spliceIndex,
+              _isCurrentBlockEmpty ? 1 : 0,
+              ..._frag
+            )
+            draft.resetState = true
```

Comment:
What if we used our existing `operations` queue for this? So something like:
```
draft.operations.reloadAll = true
```

Reply:
`draft.operations` is an array.